src/test/codewhisperer/commands/basicCommands.test.ts

```diff
@@ -138,139 +138,160 @@ describe('CodeWhisperer-basicCommands', function () {
-        assert.strictEqual(actual, 'Failed to save editor text changes into the file.')
+        it('handles document write failure', async function () {
+            const fileName = 'sample.py'
+            saveStub.resolves(true)
+            const textDocumentMock = new MockDocument('first line\n second line\n fourth line', fileName, saveStub)
 
-        sandbox.restore()
-    })
-    it('handles document write failure', async function () {
-        const fileName = 'sample.py'
-        const saveStub = sinon.stub()
-        saveStub.resolves(true)
-        const textDocumentMock = new MockDocument('first line\n second line\n fourth line', fileName, saveStub)
-
-        const openTextDocumentMock = sinon.stub()
-        openTextDocumentMock.resolves(textDocumentMock)
-        const writeFileMock = sinon.stub()
-        writeFileMock.rejects('writing to file failed.')
-
-        const sandbox = sinon.createSandbox()
-        sandbox.stub(vscode.workspace, 'openTextDocument').value(openTextDocumentMock)
-        sinon.stub(FileSystemCommon.prototype, 'writeFile').value(writeFileMock)
-        const loggerStub = sinon.stub(getLogger(), 'error')
-
-        targetCommand = testCommand(applySecurityFix)
-        await targetCommand.execute({
-            filePath: fileName,
-            ...createCodeScanIssue({
+            openTextDocumentMock.resolves(textDocumentMock)
+            writeFileMock.rejects('Error: Writing to file failed.')
+
+            sandbox.stub(vscode.workspace, 'openTextDocument').value(openTextDocumentMock)
+            sinon.stub(FileSystemCommon.prototype, 'writeFile').value(writeFileMock)
+            const loggerStub = sinon.stub(getLogger(), 'error')
+
+            targetCommand = testCommand(applySecurityFix)
+            const codeScanIssue = createCodeScanIssue({
                 suggestedFixes: [
                     {
                         description: 'fix',
                         code: '@@ -1,3 +1,3 @@\n first line\n- second line\n+ third line\n  fourth line',
                     },
                 ],
-            }),
-        })
-
-        assert.ok(saveStub.calledOnce)
-        assert.ok(loggerStub.calledOnce)
-        const actual = loggerStub.getCall(0).args[0]
-        assert.strictEqual(actual, 'Unable to write updated text content into the file: writing to file failed.')
+            })
+            await targetCommand.execute(codeScanIssue, fileName, 'hover')
 
-        sandbox.restore()
+            assert.ok(saveStub.calledOnce)
+            assert.ok(loggerStub.calledOnce)
+            const actual = loggerStub.getCall(0).args[0]
+            assert.strictEqual(actual, 'Apply fix command failed. Error: Writing to file failed.')
+            assertTelemetry('codewhisperer_codeScanIssueApplyFix', {
+                detectorId: codeScanIssue.detectorId,
+                findingId: codeScanIssue.findingId,
+                applyFixSource: 'hover',
```

Comment:
nit: can we use different source values (`hover`, `webview`, or `quickfix`) to verify that `applyFixSource` value is not hardcoded by any chance.